### PR TITLE
return more struct info

### PR DESCRIPTION
**Pull request by rcannood**
## Issue ticket number and link
Closes #xxxx (Replace xxxx with the GitHub issue number)

## Checklist before requesting a review
- [ ] I have performed a self-review of my code

- Check the correct box. Does this PR contain:
  - [x] Breaking changes
  - [x] New functionality
  - [x] Major changes
  - [ ] Minor changes
  - [ ] Bug fixes

- [ ] Proposed changes are described in the CHANGELOG.md

- [ ] CI Tests succeed and look good!

### CODE REVIEW

**Comment by rcannood** [comment]
Maybe it would be nice if the output stored in `struct` is more or less in line with the format in `api/file_*.yaml` files, that is:

```yaml
type: file
example: "resources_test/common/pancreas/raw.h5ad"
info:
  label: "Raw dataset"
  summary: An unprocessed dataset as output by a dataset loader.
  description: |
    This dataset contains raw counts and metadata as output by a dataset loader.

    The format of this file is derived from the [CELLxGENE schema v4.0.0](https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/4.0.0/schema.md).
  slots:
    layers: 
      - type: integer
        name: counts
        description: Raw counts
        required: true
    obs:
      - type: string
        name: dataset_id
        description: Identifier for the dataset from which the cell data is derived, useful for tracking and referencing purposes.
        required: false

      - type: string
        name: assay
        description: Type of assay used to generate the cell data, indicating the methodology or technique employed.
        required: false

      - type: string
        name: assay_ontology_term_id
        description: Experimental Factor Ontology (`EFO:`) term identifier for the assay, providing a standardized reference to the assay type.
        required: false

      - type: string
        name: cell_type
        description: Classification of the cell type based on its characteristics and function within the tissue or organism.
        required: false

      - type: string
        name: cell_type_ontology_term_id
        description: Cell Ontology (`CL:`) term identifier for the cell type, offering a standardized reference to the specific cell classification.
        required: false

      - type: string
        name: development_stage
        description: Stage of development of the organism or tissue from which the cell is derived, indicating its maturity or developmental phase.
        required: false

      - type: string
        name: development_stage_ontology_term_id
        description: |
          Ontology term identifier for the developmental stage, providing a standardized reference to the organism's developmental phase.

          If the organism is human (`organism_ontology_term_id == 'NCBITaxon:9606'`), then the Human Developmental Stages (`HsapDv:`) ontology is used.  
          If the organism is mouse (`organism_ontology_term_id == 'NCBITaxon:10090'`), then the Mouse Developmental Stages (`MmusDv:`) ontology is used.
          Otherwise, the Uberon (`UBERON:`) ontology is used.
        required: false

      - type: string
        name: disease
        description: Information on any disease or pathological condition associated with the cell or donor.
        required: false

      - type: string
        name: disease_ontology_term_id
        description: |
          Ontology term identifier for the disease, enabling standardized disease classification and referencing.

          Must be a term from the Mondo Disease Ontology (`MONDO:`) ontology term, or `PATO:0000461` from the Phenotype And Trait Ontology (`PATO:`).
        required: false

      - type: string
        name: donor_id
        description: Identifier for the donor from whom the cell sample is obtained.
        required: false

      - type: boolean
        name: is_primary_data
        description: Indicates whether the data is primary (directly obtained from experiments) or has been computationally derived from other primary data.
        required: false

      - type: string
        name: organism
        description: Organism from which the cell sample is obtained.
        required: false
      
      - type: string
        name: organism_ontology_term_id
        description: |
          Ontology term identifier for the organism, providing a standardized reference for the organism.

          Must be a term from the NCBI Taxonomy Ontology (`NCBITaxon:`) which is a child of `NCBITaxon:33208`.
        required: false

      - type: string
        name: self_reported_ethnicity
        description: Ethnicity of the donor as self-reported, relevant for studies considering genetic diversity and population-specific traits.
        required: false

      - type: string
        name: self_reported_ethnicity_ontology_term_id
        description: |
          Ontology term identifier for the self-reported ethnicity, providing a standardized reference for ethnic classifications.

          If the organism is human (`organism_ontology_term_id == 'NCBITaxon:9606'`), then the Human Ancestry Ontology (`HANCESTRO:`) is used.
        required: false

      - type: string
        name: sex
        description: Biological sex of the donor or source organism, crucial for studies involving sex-specific traits or conditions.
        required: false

      - type: string
        name: sex_ontology_term_id
        description: Ontology term identifier for the biological sex, ensuring standardized classification of sex. Only `PATO:0000383`, `PATO:0000384` and `PATO:0001340` are allowed.
        required: false

      - type: string
        name: suspension_type
        description: Type of suspension or medium in which the cells were stored or processed, important for understanding cell handling and conditions.
        required: false

      - type: string
        name: tissue
        description: Specific tissue from which the cells were derived, key for context and specificity in cell studies.
        required: false

      - type: string
        name: tissue_ontology_term_id
        description: |
          Ontology term identifier for the tissue, providing a standardized reference for the tissue type.

          For organoid or tissue samples, the Uber-anatomy ontology (`UBERON:`) is used. The term ids must be a child term of `UBERON:0001062` (anatomical entity).
          For cell cultures, the Cell Ontology (`CL:`) is used. The term ids cannot be `CL:0000255`, `CL:0000257` or `CL:0000548`.
        required: false

      - type: string
        name: tissue_general
        description: General category or classification of the tissue, useful for broader grouping and comparison of cell data.
        required: false

      - type: string
        name: tissue_general_ontology_term_id
        description: |
          Ontology term identifier for the general tissue category, aiding in standardizing and grouping tissue types.

          For organoid or tissue samples, the Uber-anatomy ontology (`UBERON:`) is used. The term ids must be a child term of `UBERON:0001062` (anatomical entity).
          For cell cultures, the Cell Ontology (`CL:`) is used. The term ids cannot be `CL:0000255`, `CL:0000257` or `CL:0000548`.
        required: false

      - type: string
        name: batch
        description: A batch identifier. This label is very context-dependent and may be a combination of the tissue, assay, donor, etc.
        required: false

      - type: integer
        name: soma_joinid
        description: If the dataset was retrieved from CELLxGENE census, this is a unique identifier for the cell.
        required: false
    var:
      - type: string
        name: feature_id
        description: Unique identifier for the feature, usually a ENSEMBL gene id.
        required: false
      
      - type: string
        name: feature_name
        description: A human-readable name for the feature, usually a gene symbol.
        required: false

      - type: integer
        name: soma_joinid
        description: If the dataset was retrieved from CELLxGENE census, this is a unique identifier for the feature.
        required: false
    uns:
      - type: string
        name: dataset_id
        description: A unique identifier for the dataset. This is different from the `obs.dataset_id` field, which is the identifier for the dataset from which the cell data is derived.
        required: true
      - name: dataset_name
        type: string
        description: A human-readable name for the dataset.
        required: true
      - type: string
        name: dataset_url
        description: Link to the original source of the dataset.
        required: false
      - name: dataset_reference
        type: string
        description: Bibtex reference of the paper in which the dataset was published.
        required: false
        multiple: true
      - name: dataset_summary
        type: string
        description: Short description of the dataset.
        required: true
      - name: dataset_description
        type: string
        description: Long description of the dataset.
        required: true
      - name: dataset_organism
        type: string
        description: The organism of the sample in the dataset.
        required: false
        multiple: true
```

→ This would require making a distinction between type and dtype in the `file_*.yaml` where there currently is none.
→ Would require reformatting the struct object returned by this component

File: src/common/extract_metadata/config.vsh.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  name: extract_metadata
+  namespace: common
+  description: Extract the metadata from an h5ad file.
+  argument_groups:
+    - name:  Inputs
+      arguments:
+        - name: --input
+          type: file
+          required: true
+          description: A h5ad file.
```
Comment (rcannood) [inline src/common/extract_metadata/config.vsh.yaml:11]:
```suggestion
        - name: --input
          type: file
          required: true
          description: A h5ad file.
        - name: --schema
          type: file
          required: false
          description: An optional schema with which to annotate the output
```

↑ What would you think about adding an optional `schema` so we can add descriptions to the fields that are in the AnnData and also in the schema?

Comment (KaiWaldrant) [reply src/common/extract_metadata/config.vsh.yaml:11]: ok for me

Comment (rcannood) [reply src/common/extract_metadata/config.vsh.yaml:11]: Working on it!

Comment (rcannood) [reply src/common/extract_metadata/config.vsh.yaml:11]: Done!

**Pull request merged**